### ISSUE TITLE
ctfe public key format is not limited to ECDSA

### ISSUE THREAD

**Issue by loosebazooka** [sigstore/sigstore-java#4]
Can be ECDSA, RSA or EDDSA/ED25519, handle this appropriately when loading the key for verification.

https://github.com/sigstore/sigstore-java/blob/main/src/main/java/dev/sigstore/fulcio/client/FulcioValidator.java#L50

**Comment by loosebazooka** [comment]
Ideally we would have something utility that can parse these things for us that we can use all over the client.
```
public class Keys {
  public static PublicKey parseKey(byte[] keyBytes);
  public static X509Certificate parseCertificate(byte[] certificateBytes);
}
```

**Comment by loosebazooka** [comment]
fixed by #13
